packages/babel-helpers/src/helpers.js

```diff
@@ -433,7 +433,7 @@ helpers.objectDestructuringEmpty = template(`
 
 helpers.objectWithoutProperties = template(`
   (function (source, excluded) {
-    if (source === null) return {};
+    if (source === null) return null;
```

Comment:
Aargh. I'm not sure why I'm making silly mistakes.

Also it'd still be `return {}` because it returns target and target is a new object [spec, note 2 and step 6](https://tc39.github.io/proposal-object-rest-spread/#AbstractOperations-CopyDataProperties)